cedar-policy/src/ffi/is_authorized.rs

```diff
@@ -352,24 +351,27 @@ impl TryFrom<crate::PartialResponse> for ResidualResponse {
             decision: partial_response.decision(),
             satisfied: partial_response
                 .definitely_satisfied()
-                .map(|p| p.id().clone())
+                .map(|p| p.id().to_smolstr())
```

Comment:
Not sure if `.to_smolstr()` is smart enough to realize that `PolicyId` just wraps `SmolStr` itself (through a couple layers), and just clone the `SmolStr` here (O(1)) instead of using `Display` and allocating a new `SmolStr`.  Do we care?  Alternative is to make a dedicated method on `PolicyId` that does the right thing, even if only `pub(crate)`

Reply:
I added a `pub(crate)` `into_smolstr` function in the latest commit, which helps get rid of one clone. It's not useful in this particular case though since the id comes from `Policy::id()`, which necessarily returns a reference, since under the hood policies store an `Arc<Template>`. There needs to be a clone somewhere along the call chain, and it seems reasonable to make it here.

Reply:
Makes sense.